### PR TITLE
PHP: Refactor helper to reduce code duplication

### PR DESCRIPTION
**Pull request by LeoColomb**
While working on #1039 I faced some 💢😤💥🤬 for the `Installer` instantiation because it was duplicated between `getLatestResolvableVersion` and `update` functions.
Nothing functional is changed, only code organization to match with common practices. 🙂

In the meantime, I switched linter from php-cs-fixer to phpcs that do not require local config neither `--dry-run` for the exact same result.

### CODE REVIEW

**Comment by LeoColomb** [comment]
This can help to avoid situations like https://github.com/dependabot/dependabot-core/commit/a926bb2310eca3e14f0ede3a1f530c9f04f451e1! 😇 😄 😇

File: composer/helpers/bin/run [hunk 1]
```diff
@@ -5,15 +5,15 @@ declare(strict_types=1);
 
 namespace Dependabot\PHP;
 
-require __DIR__ . '/../vendor/autoload.php';
+require dirname(__DIR__) . '/vendor/autoload.php';
```
[the remark this answers is not in the document]
Comment (LeoColomb) [reply composer/helpers/bin/run]: Switching from string implicitness to function explicitness.

File: composer/helpers/src/Helper.php [hunk 2]
```diff
@@ -5,77 +5,83 @@
...
 use Composer\Factory;
-use Composer\Installer;
 use Composer\Package\PackageInterface;
 
-class UpdateChecker
+class Helper
 {
+    public static function getContentHash(array $args): ?string
+    {
+        [$workingDirectory] = $args;
```
[the remark this answers is not in the document]
Comment (LeoColomb) [reply composer/helpers/src/Helper.php]: > Perhaps this is out of scope of this pull request, though.

Indeed! 😁
Perfection is not the goal, only DRY and standard compliance.

File: composer/helpers/bin/run [hunk 3]
```diff
@@ -29,7 +29,7 @@ if (function_exists('ini_set')) {
 
     $memoryLimit = trim(ini_get('memory_limit'));
     // Increase memory_limit if it is lower than 1900MB
-    if ($memoryLimit != -1 && $memoryInBytes($memoryLimit) < 1024 * 1024 * 1900) {
+    if (!empty($memoryLimit) && $memoryInBytes($memoryLimit) < 1024 * 1024 * 1900) {
```
[the remark this answers is not in the document]
Comment (LeoColomb) [reply composer/helpers/bin/run]: > What do you think about avoiding usage of empty() here, and instead comparing with an empty string?

What's the point?

File: composer/helpers/src/Helper.php [hunk 4]
```diff
@@ -5,77 +5,83 @@
 namespace Dependabot\PHP;
 
 use Composer\Factory;
-use Composer\Installer;
 use Composer\Package\PackageInterface;
 
-class UpdateChecker
+class Helper
```
[the remark this answers is not in the document]
Comment (LeoColomb) [reply composer/helpers/src/Helper.php]: I don't think a constrictor can be private. 

Why not for the `final`.

File: composer/helpers/bin/run [hunk 5]
```diff
@@ -5,15 +5,15 @@ declare(strict_types=1);
...
+require dirname(__DIR__) . '/vendor/autoload.php';
 
 // Get details of the process to run from STDIN. It will have a `function`
 // and an `args` method, as passed in by UpdateCheckers::Php
 $request = json_decode(file_get_contents('php://stdin'), true);
 
 // Increase the default memory limit the same way Composer does (but clearer)
 if (function_exists('ini_set')) {
-    $memoryInBytes = function ($value) {
+    $memoryInBytes = function (string $value): int {
```
[the remark this answers is not in the document]
Comment (LeoColomb) [reply composer/helpers/bin/run]: ?

File: composer/helpers/bin/run (hunk 3 above)
[the remark this answers is not in the document]
Comment (localheinz) [reply composer/helpers/bin/run]: The point is to be more strict: the function `trim()` returns a `string` in any case, so use of the generic `empty()` is  not needed.

File: composer/helpers/src/Helper.php (hunk 4 above)
[the remark this answers is not in the document]
Comment (localheinz) [reply composer/helpers/src/Helper.php]: See https://3v4l.org/njTVq.

File: composer/helpers/bin/run (hunk 3 above)
[the remark this answers is not in the document]
Comment (LeoColomb) [reply composer/helpers/bin/run]: OK, I see. What do you suggest?
I can switch to 
```suggestion
    if ($memoryLimit !== "" && $memoryInBytes($memoryLimit) < 1024 * 1024 * 1900) {
```

That said, I'm not sure about its readability and strictness.

File: composer/helpers/src/Helper.php (hunk 4 above)
[the remark this answers is not in the document]
Comment (LeoColomb) [reply composer/helpers/src/Helper.php]: Oh great! Alright.

[the remark this answers is not in the document]
Comment (LeoColomb) [reply composer/helpers/src/Helper.php]: Updated.

File: composer/helpers/bin/run (hunk 5 above)
[the remark this answers is not in the document]
Comment (LeoColomb) [reply composer/helpers/bin/run]: Updated

File: composer/helpers/bin/run (hunk 3 above)
[the remark this answers is not in the document]
Comment (localheinz) [reply composer/helpers/bin/run]: 👍

[the remark this answers is not in the document]
Comment (LeoColomb) [reply composer/helpers/bin/run]: Fixed

File: composer/helpers/src/Helper.php (hunk 2 above)
[the remark this answers is not in the document]
Comment (LeoColomb) [reply composer/helpers/src/Helper.php]: @localheinz I actually reflect your suggestion in 6790e64. Would love to have your review! 🙂

**Comment by localheinz** [comment]
@LeoColomb 

Looks like the build failure is unrelated to your changes (not entirely sure, though). Since we have conflicts, what do you think about resolving them and pushing the changes to trigger another build?

**Comment by LeoColomb** [comment]
@localheinz The build failure is unrelated, indeed. The instability of build result has already been reported.

That said, as you may see I already rebased my PR many times (16 force-push are reported on the log), and even if I like to work with git I cannot rebase each time `composer.lock` receive an update during half a year. I'm not dedicated to dependabot! 😉

**Comment by localheinz** [comment]
@LeoColomb 

No problem, I would not mind opening a new PR based on your work, and then see it through - what do you think?

**Comment by LeoColomb** [comment]
> I would not mind

Do you mean "I would mind"? If so, sure, feel free! 🙂 
If not, sorry, I don't understand your sentence... 😇

**Comment by localheinz** [comment]
@LeoColomb 

I meant that - unless you have objections - I will be happy to continue the work you started by opening one or more pull requests, based on or inspired by your work.

👍

**Comment by LeoColomb** [comment]
@localheinz I don't have any objection! 😉  
That said, I would prefer too see this PR merged or understand why it is not.

**Comment by LeoColomb** [comment]
@greysteil @localheinz Any news here?

**Pull request closed without merging**